homeassistant/components/mqtt/client.py

```diff
@@ -660,6 +662,8 @@ async def _register_mid(self, mid: int) -> None:
         async with self._pending_operations_condition:
             if mid not in self._pending_operations:
                 self._pending_operations[mid] = asyncio.Event()
+                # Late ACK registrations might come while unloading the integration.
+                self._pending_operations_condition.notify_all()
```

Comment:
Make sure the new registered event is noticed during unloading. It seems this extra line makes the warnings go away during an unload.